lib/spack/spack/ci.py

```diff
@@ -1289,33 +1271,16 @@ def main_script_replacements(cmd):
                         ]
                     )
 
-                job_object = {
-                    "stage": stage_name,
-                    "variables": variables,
-                    "script": job_script,
-                    "tags": tags,
-                    "artifacts": {"paths": artifact_paths, "when": "always"},
-                    "needs": sorted(job_dependencies, key=lambda d: d["job"]),
-                    "retry": {"max": 2, "when": JOB_RETRY_CONDITIONS},
-                    "interruptible": True,
-                }
+                job_object["stage"] = stage_name
+                job_object["needs"] = sorted(job_object["needs"], key=lambda d: d["job"])
```

Comment:
Is there a compelling reason to/not to sort these? Not asking to change it, just curious.

Reply:
The original code sorted them, I didn't change that. AFAIK the order makes no difference from GitLab's perspective.

Reply:
Went ahead and removed this sort.